sdk/typescript/src/types/framework.ts

```diff
@@ -16,11 +16,19 @@ export class Coin {
     return getMoveObjectType(data)?.startsWith('0x2::Coin::Coin') ?? false;
   }
 
+  static convertToBN(balance : string) {
+    return new BN.BN(balance, 10);
+  }
+
   static getBalance(data: GetObjectInfoResponse): BN | undefined {
     if (!Coin.isCoin(data)) {
       return undefined;
     }
     const balance = getObjectFields(data)?.balance;
-    return new BN.BN(balance, 10);
+    return this.convertToBN(balance);
```

Comment:
You can testing this change with `yarn start` which connects to the DevNet and navigate to the page http://localhost:3000/addresses/0x813ec1ab1e1797c79d1e1fcbebebc27a1fd21f07. You will see the amount failed to load and a console error. If you convert `balance` into `String(balance)`, the error will be gone but it will display an incorrect amount for each coin (`18446744073709552000` instead of `18446744073709551615`). You can obtain the latter result through this curl command

```
curl --location --request POST 'https://gateway.devnet.sui.io:443' \
--header 'Content-Type: application/json' \
--data-raw '{
    "jsonrpc":"2.0",
    "id":1,
    "method":"sui_getObjectInfo",
    "params":["0x003e1d8b91a4cb1538cdb50c3cae8e62e1960b1c"]
}'
```

The root cause is `balance` is defined as a number on the server side and the default JSON Parser does not work well with big numbers: https://github.com/0xcert/framework/issues/453.

To fix this, I think you would need to use https://github.com/josdejong/lossless-json#bignumbers to parse the JSON object in https://github.com/MystenLabs/sui/blob/459ec5e699f93a793649a5c49a06d69cc5e4c312/sdk/typescript/src/rpc/client.ts#L47 

An alternative way is to change the gateway implementation to represent numbers in string. Ideally we don't want to take this approach since it results in loss of information and the root cause is in the javascript JSON parser layer.

Reply:
At some point the large integer causing the overflow problem will have to be converted to a string in order for JavaScript to display it without error. The loss of information seems minimal as this process involves going from some large number ( e.g. 99999999999999999999) to a series of characters (e.g. "99999999999999999999"). My assumption, is that we had pursued this typical, neatest solution, which is to have the conversion occur in the RPC API server, as then web developers using languages other than TypeScript (e.g. Python, Java or Ruby)  would not be caught off guard by the overflow issue.

Ideally, the gateway implementation should be changed to represent numbers as strings, which the recommended solution from [0xcert](https://github.com/0xcert/framework/issues/453#:~:text=JSON%20%28JavaScript%20Object%20Notation%29%20should%20represent%20numbers%20the,used%20in%20the%20framework.%20That%20is%20a%20workaround).

Reply:
> An alternative way is to change the gateway implementation to represent numbers in string. Ideally we don't want to take this approach since it results in loss of information and the root cause is in the javascript JSON parser layer.

As I mentioned above, this results in loss of information(i.e., types) for arbitrary move object and is not ideal. @patrickkuo what do you think?

Reply:
OK, so, for now, to turn this into an operational PR, I've created a fix that works with the problematic `0x003e1d8b91a4cb1538cdb50c3cae8e62e1960b1c` using the suggested `lossless-json` library. This approach requires specifying which fields are likely to have the overflow issue; here we specify just the `balance` field.

Reply:
I think the type information is partly lost already when we convert to JSON, e.g. we can't distinguish between u8/u64... u128 is already a String, I guess it's ok to use String to ensure safety.

@666lcz are there any other implication apart from type info loss if we use String to represent all numbers?

Reply:
we want to use hex String to convey arbitrarily large numbers - this is how most of blockchain does it.